filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
+	m.mu.Unlock()
```

Comment:
hmm what if you call `Close()` after `Unlock()`? The go docs say of `sync.Mutex`:

> It is a run-time error if m is not locked on entry to Unlock

It seems to me that the `mu` member is not even needed - perhaps just remove it entirely in this pr?

Reply:
I agree with @alexflint that we should just drop the `mu` member.  The existing file locking semantics are strictly stronger than those of `mu`.  It would then simplify these things a lot.